network/src/network.rs

```diff
@@ -1521,18 +1523,33 @@ pub(crate) async fn async_disconnect_with_message(
 }
 
 #[derive(Clone, Copy, Debug, Eq, PartialEq, PartialOrd, Ord)]
-pub(crate) enum TransportType {
+pub enum TransportType {
     Tcp,
     Ws,
+    Wss,
 }
 
-pub(crate) fn find_type(addr: &Multiaddr) -> TransportType {
-    if addr
-        .iter()
-        .any(|proto| matches!(proto, Protocol::Ws | Protocol::Wss))
-    {
-        TransportType::Ws
-    } else {
-        TransportType::Tcp
+impl<'a> From<TransportType> for p2p::multiaddr::Protocol<'a> {
+    fn from(value: TransportType) -> Self {
+        match value {
+            TransportType::Ws => Protocol::Ws,
+            TransportType::Wss => Protocol::Wss,
+            _ => unreachable!(),
```

Comment:
Is this possible that a match arm is a `TransportType::Tcp`?

Reply:
First, there is no such scenario in the current code. 
Second, it is impossible to convert here because TCP needs to contain port information, but the type cannot know it.

This conversion is only performed when OutboundPeerService dial

Reply:
If it is only used inside the OutboundPeerService dial fn, it is recommended to use a more intuitive way to write the conversion in the dial fn:
```rust
        for mut addr in peers {
            match &self.transport_type {
                TransportType::Tcp => (),
                TransportType::Ws => addr.push(Protocol::Ws),
                TransportType::Wss => addr.push(Protocol::Wss)
            }

            self.network_state.dial_identify(&self.p2p_control, addr);
        }
```